app/src/main/java/chat/rocket/android/push/PushNotificationHandler.java

```diff
@@ -0,0 +1,654 @@
+      messageMap.put(notId, messageList);
+    }
+
+    if (message.isEmpty()) {
+      messageList.clear();
+    } else {
+      messageList.add(message);
+    }
+  }
+
+  public void showNotificationIfPossible(Context context, PushInteractor pushInteractor,
+                                         Bundle extras) {
+
+    // Send a notification if there is a message or title, otherwise just send data
+    String message = extras.getString(MESSAGE);
+    String title = extras.getString(TITLE);
+    String contentAvailable = extras.getString(CONTENT_AVAILABLE);
+    String forceStart = extras.getString(FORCE_START);
+
+    Log.d(LOG_TAG, "message =[" + message + "]");
+    Log.d(LOG_TAG, "title =[" + title + "]");
+    Log.d(LOG_TAG, "contentAvailable =[" + contentAvailable + "]");
+    Log.d(LOG_TAG, "forceStart =[" + forceStart + "]");
+
+    if ((message != null && message.length() != 0) ||
+        (title != null && title.length() != 0)) {
+
+      Log.d(LOG_TAG, "create notification");
+
+      if (title == null || title.isEmpty()) {
+        extras.putString(TITLE, getAppName(context));
+      }
+
+      createNotification(context, pushInteractor, extras);
+    }
+  }
+
+  public void createNotification(Context context, PushInteractor pushInteractor, Bundle extras) {
+    NotificationManager mNotificationManager =
+        (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+    String appName = getAppName(context);
+    String packageName = context.getPackageName();
+    Resources resources = context.getResources();
+
+    String serverUrl = getServerUrl(extras);
+    String roomId = getRoomId(extras);
+
+    if (serverUrl == null || roomId == null) {
+      return;
+    }
+
+    String serverConfigId = pushInteractor.getServerConfigId(serverUrl);
+    if (serverConfigId == null) {
+      return;
+    }
+
+    int notId = parseInt(NOT_ID, extras);
+    Intent notificationIntent = new Intent(context, MainActivity.class);
+    notificationIntent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+    notificationIntent.putExtra(PUSH_BUNDLE, extras);
+    notificationIntent.putExtra(SERVER_CONFIG_ID, serverConfigId);
+    notificationIntent.putExtra(ROOM_ID, roomId);
+    notificationIntent.putExtra(NOT_ID, notId);
+
+    int requestCode = random.nextInt();
+    PendingIntent contentIntent = PendingIntent
+        .getActivity(context, requestCode, notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+    NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context)
+        .setWhen(System.currentTimeMillis())
+        .setContentTitle(fromHtml(extras.getString(TITLE)))
+        .setTicker(fromHtml(extras.getString(TITLE)))
+        .setContentIntent(contentIntent)
+        .setAutoCancel(true);
+
+    SharedPreferences prefs = context
+        .getSharedPreferences(PushConstants.COM_ADOBE_PHONEGAP_PUSH, Context.MODE_PRIVATE);
```

Comment:
The Cordova plugin uses this for some of its configuration. Since we'll have some white label work I thought that it would be nice to leave it there.